### PR TITLE
Use name to lookup graphql by operation name

### LINKED ISSUE #334: Feedback on graphClient.executeQueryFromFile versus graphClient.query
```
ctx.graphClient.executeQueryFromFile<PodDeployments.PodDeployment, PodDeployments.Variables>(
        "./src/graphql/query/podDeployments.graphql",
        {
        });
```
versus

```
ctx.graphClient.query<PodDeployments.PodDeployment, PodDeployments.Variables>(
       {name: "podDeployments",
        options: {}}
        );
```

I think the latter is preferred, where you use the convention for how graphql query files are laid out in the underlying node module.  

There is still a potential difference between the name of the graphql file and and the name of the graphql query in that file.  For some reason, I was sort of expecting those graphql files to have been indexed by the  names of the queries they contain so it was only on my second run through that I tried using the query's filename.  

I also had an issue where I had a mutation graphql in the query directory and then had some difficult figuring out that I needed to use the `mutation` api, and move the file into the other structure.  The convention seems good but I think there are some important missing docs here.  The GraphClient interface mentions " Run query either provided as a string or as loaded from a file" but the QueryOptions doesn't have enough context to figure out this convention. 

The exmaples here:  https://docs.atomist.com/developer/graphql/ do not refer to the `query` api so also wnat to confirm that we should be using this one.

### PR DESCRIPTION
**Pull request by cdupuis**
fixes #334

### CODE REVIEW

File: src/spi/graph/GraphClient.ts [hunk 1]
```diff
@@ -63,28 +60,63 @@ export interface GraphClient {
      * @param {Q} variables the variables to be used
      * @param {any} options to be passed onto the underlying implementation
      * @returns {Promise<T>}
-     *
-     * DEPRECATED: use mutate() instead
+     * @deprecated use mutate() instead
      */
     executeMutation<T, Q>(mutation: string, variables?: Q, options?: any): Promise<any>;
```
Comment (ddgenome) [inline src/spi/graph/GraphClient.ts]: I know you didn't change this, but shouldn't this return `Promise<T>`?

File: src/internal/graph/graphQL.ts [hunk 2]
```diff
@@ -343,12 +344,30 @@ function locateAndLoadGraphql(
...
+                logger.warn(`WARNING: no ${subfolder} graphql operation found for name '${options.name}. ` +
+                    `Falling back to file name lookup. Support for file name lookup will be removed in a future release.'`);
+                if (!options.name.endsWith(".graphql")) {
+                    path = p.join(queryDir, `${options.name}.graphql`);
+                } else {
+                    path = p.join(queryDir, options.name);
+                }
+                // End of remove for next major release
+                // throw new Error(`No matching ${subfolder} graphql operation found for name '${options.name}'`);
+            } else if (queries.length > 1) {
```
Comment (ddgenome) [inline src/internal/graph/graphQL.ts]: `else` would probably suffice here.

File: src/spi/graph/GraphClient.ts [hunk 3]
```diff
@@ -63,28 +60,63 @@ export interface GraphClient {
...
+
+    /** Optional GraphQL query string */
     query?: string;
+    /**
+     * Optional absolute or relative path to the '.graphql' query file.
+     * Relative paths will get resolved against the path of the calling script.
+     */
     path?: string;
+    /**
+     * Optional name of the query operation in a '.graphql' file located in 'graphql/query'.
```
Comment (ddgenome) [inline src/spi/graph/GraphClient.ts]: One must specify either `query` or `name`, correct?

File: src/spi/graph/GraphClient.ts [hunk 4]
```diff
@@ -63,28 +60,63 @@ export interface GraphClient {
...
+
+    /** Optional GraphQL mutation string */
     mutation?: string;
+    /**
+     * Optional absolute or relative path to the '.graphql' query file.
+     * Relative paths will get resolved against the path of the calling script.
+     */
     path?: string;
+    /**
+     * Optional name of the mutation operation in a '.graphql' file located in 'graphql/mutation'.
```
Comment (ddgenome) [inline src/spi/graph/GraphClient.ts]: One of `mutation` or `query` must be provided, correct?

Comment (cdupuis) [reply src/spi/graph/GraphClient.ts]: Add a interface-level comment to reflect that.

File: src/spi/graph/GraphClient.ts (hunk 3 above)
Comment (cdupuis) [reply src/spi/graph/GraphClient.ts]: Add a interface-level comment to reflect that.

**Pull request merged**